src/hotspot/share/gc/serial/cardTableRS.cpp

```diff
@@ -601,20 +344,187 @@ void CardTableRS::verify() {
 CardTableRS::CardTableRS(MemRegion whole_heap) :
   CardTable(whole_heap) { }
 
+// Implemented word-iteration to skip long consecutive clean cards.
+CardTable::CardValue* CardTableRS::find_first_dirty_card(CardValue* const start_card,
+                                                         CardValue* const end_card) {
+  using Word = uintptr_t;
+
+  auto is_word_aligned = [] (const void* const p) {
+      return (((uintptr_t)p) % sizeof(Word)) == 0;
+  };
+
+  static_assert(clean_card_val() == (CardValue)-1, "inv");
+  constexpr Word clean_word = (Word)-1;
+
+  CardValue* i = start_card;
```

Comment:
I don't much like naming a pointer variable "i".  "i" is a really commonly used index variable name.